Feature/task creds changes

kingy04 opened the pull request:
A couple of CSS changes made for [CC-1551](https://croudcontrol.atlassian.net/browse/CC-1551)

- Fixes an issue on the summarised dropdown with large options arrays, the dropdown would be very long and show the whole list rather than a scrollable reasonable size list like other select dropdowns.. i have tested the tweak in each SPA the dropdown is used in the monorepo

- Adds a media query to the sidebar component, which can optionally be used by adding a '.responsive' class to a sidebar, which will set the width of the sidebar to 100% minus the CC menu width if it reaches 768px or below.. this is a better solution than applying these tweaks in each SPA and is a nicer UX

Commit:
remove css causing as overly long dropdown if options array is large

Commit:
css tweak for the universal sidebar

BrockReece commented on src/components/shared/layout/Sidebar.vue:
```diff
@@ -144,4 +146,12 @@
+
+    @media (max-width: 768px) {
+
+        .ui.segments.universal-editor.responsive {
```
Should this be on by default if it is a UX concern?
Could possibly have a responsive prop that was true by default so it can be overridden in special cases?

BrockReece replied:
What is the offset logic and where is it used?

kingy04 replied:
Essentially its the 55px sidebar width which we are offsetting here `$full-screen-breakpoint: $sidebar-max-width + $collapsed-cc-menu-width;`

I have added an additional .no-offset class, in this case it just sets the width to 100vw

Thanks

BrockReece reviewed (commented):
Just a few comments on the changes. 
Are we supporting mobile browsers now?

kingy04 commented:
I don't think we are supporting mobile browsers in v2, this change keeps content on the page rather than allowing it to escape the page like so

![Uploading Screen Shot 2018-08-17 at 9.59.30 am.png…]()

BrockReece commented on src/components/shared/layout/Sidebar.vue:
```diff
@@ -144,4 +146,12 @@
         }
     }
+
+    @media (max-width: 768px) {
```
Wouldn't it be 755px? as in `$collapsed-cc-menu-width` (55px) + `.ui.segments.universal-editor`'s width (700px). Either way, if it is a width specific to this component, it would be useful to use another sass variable here so it doesn't seem so arbitrary. 

```scss
$sidebar-max-width: 700px;
$collapsed-cc-menu-width: 55px;
$full-screen-breakpoint: $sidebar-max-width + $collapsed-cc-menu-width;
...
@media (max-width: $full-screen-breakpoint) {
...

```

BrockReece commented:
Have you got an example of this causing an issue on desktop browsers?

Commit:
improve sass variable usage,   add no-offset class

Commit:
grammar

BrockReece commented:
Ok, so it is only when a user resizes the browser to a sub mobile width that there is an issue. So I think my earlier point still stands that if this is a genuine global UX concern, this should be responsive by default. Why would we want only some of our sidebars to be responsive?

This will need to go through Product before it is merged in so I will need to build a case for this. It will be hard to argue that this issue is big enough to fix but not enough of a concern to enforce the fix on all sidebars by default.

Do we have a Jira ticket for this?

paul-watkins-croud commented:
@kingy04 Can you liaise with product directly on this. Really these should be identified and signed off prior to making changes and PRing them, with the product feedback explained in the PR. There's a risk of carrying out redundant work otherwise.

This way @BrockReece only needs to be concerned with code quality rather than needing to get drawn into larger discussions.

Commit:
revert sidebar changes

Commit:
enable overiding of event listeners

kingy04 commented on src/components/shared/Datatable.vue:
```diff
@@ -34,11 +34,11 @@
                         'vuetable:cell-clicked': this.onCellClicked,
                         'vuetable:loading': () => { this.loading = true },
                         'vuetable:loaded': () => { this.loading = false },
+                        ...this.$listeners,
```
This change allows me to overide the `vuetable:cell-clicked` default method which opens the detail row... in the case of task creds, i only want to show the detail row if a session exists so need more logic in the event listener method... If you prefer i can tweak this change to only allow overiding this single listener rather than all of them.. let me know

BrockReece replied:
No this makes sense. These are only here for convenience so no reason why they can't be overriden, as long as the override doesn't break the base functionality.

kingy04 commented:
Sorry for the late reply on this one guys, Good point Paul, i will talk to product going forward before making any changes.. Ryan agreed that we should apply the sidebar css change by default across V2, but has decided to postpone that change to a seperate release... I have reverted the changes to the sidebar, leaving the summarised dropdown css fix, and i have added a tweak to the datatable component to allow overriding of event listeners. Product (Ryan) has approved these changes.

BrockReece reviewed (approved):
These changes look good to me, if Product have already approved do you want me to merge this in and tag it up?

kingy04 commented:
Great, yes please Brock 👍

BrockReece commented:
Published in v.1.9.1
